### DISCUSSION TITLE
Espeak no output with Bluetooth A2DP. Need help with code.

### DISCUSSION THREAD

**Discussion by ProHarshit** [pschatzmann/arduino-espeak-ng#15] [Q&A] [answered]
Hi everyone
The code has compiled successfully and been uploaded also to ESP32 WROOM32. 
But it is not connecting to the Bluetooth speaker. 

Here is the code 
```
#include "AudioTools.h"
#include "FileSystems.h"
#include "espeak.h"
#include "AudioLibs/A2DPStream.h"

const char* name = "Active 2.0";

AudioInfo from(22050, 1, 16);
AudioInfo to(44100, 2, 16);

A2DPStream a2dp;
FormatConverterStream out(a2dp);
ESpeak espeak(out);

void setup() {
Serial.begin(115200);
espeak.begin();
audio_info espeak_info = espeak.audioInfo();
Serial.println("Starting...");
out.begin(from, to);
auto cfg = a2dp.defaultConfig(TX_MODE);
cfg.name = name;
cfg.silence_on_nodata = true;
a2dp.begin(cfg);
a2dp.setVolume(0.3);
Serial.println("A2DP Started");

void loop() {
espeak.say("Hello world!");
delay(5000);
}
```

**Comment by pschatzmann** [comment, marked as answer]
You need to do some systematic testing:
- Did you test the a2dp source with a simple test sketch ?
- Did you test the converted e-speak output ?

Here is a link to [some generic advice ](https://github.com/pschatzmann/arduino-audio-tools/wiki/It's-not-working)

ps
Don't forget to activate the logging if you want to see what a2dp is doing....
